app/components/BuyNBTC/BuyNBTC.tsx

```diff
@@ -6,38 +6,87 @@ import { WalletContext } from "~/providers/ByieldWalletProvider";
 import { ByieldWallet } from "~/types";
 import { SuiModal } from "../Wallet/SuiWallet/SuiModal";
 import { useCurrentAccount, useSignAndExecuteTransaction, useSuiClient } from "@mysten/dapp-kit";
-import { Transaction } from "@mysten/sui/transactions";
 import { useNetworkVariables } from "~/networkConfig";
 import { createBuyNBTCTxn } from "~/util/util";
-import { BUFFER_BALANCE, PRICE_PER_NBTC_IN_SUI } from "~/constant";
+import { BUY_NBTC_GAS_FEE_IN_SUI, PRICE_PER_NBTC_IN_SUI } from "~/constant";
 import { ArrowDown } from "lucide-react";
-import { useSuiBalance } from "../Wallet/SuiWallet/useSuiBalance";
-import { FormNumericInput } from "../form/FormNumericInput";
-import { Modal } from "../ui/dialog";
 import { formatSUI, parseSUI } from "~/lib/denoms";
 import { useToast } from "~/hooks/use-toast";
 import { useNBTCBalance } from "../Wallet/SuiWallet/useNBTCBalance";
 import { NBTCBalance } from "./NBTCBalance";
+import { FormNumericInput } from "../form/FormNumericInput";
+import { Modal } from "../ui/dialog";
+import { useSuiBalance } from "../Wallet/SuiWallet/useSuiBalance";
 import { Instructions } from "./Instructions";
-import { SUIGasFee } from "./SUIGasFee";
 import { TransactionStatus } from "./TransactionStatus";
 
+const calculateYouReceive = (_mistAmount: bigint, _gasFee: bigint): number => {
+	const mistAmountAfterFee = _mistAmount - _gasFee;
+	return Number(formatSUI(mistAmountAfterFee)) / PRICE_PER_NBTC_IN_SUI;
+};
+
+function SUIIcon() {
+	return (
+		<div className="flex gap-2 items-center mr-2">
+			SUI
+			<img
+				src="https://cdn.prod.website-files.com/6425f546844727ce5fb9e5ab/65690e5e73e9e2a416e3502f_sui-mark.svg"
+				loading="lazy"
+				alt="SUI"
+				className="w-7 h-7"
+			/>
+		</div>
+	);
+}
+
+function NBTCIcon() {
+	return (
+		<div className="flex gap-2 items-center mr-2">
+			nBTC
+			<img src="/nbtc.svg" alt="Bitcoin" className="w-7 h-7 mr-2" />
+		</div>
+	);
+}
```

Comment:
we should have a function that takes image as an argument, and them other functions that will shorten that:

Since this components are universal, they should be in `coponents/icons.ts`

```
function Icon(src: string) {
	return (
		<div className="flex gap-2 items-center mr-2">
			nBTC
			<img src="{src}" className="w-7 h-7 mr-2" />
		</div>
	);
}

function NBTCIcon() {
	return Icon("/nbtc.svg"
}
```

Also, `"/nbtc.svg" should be from `public/assets/nbtc.svg`

Reply:
Done